stratum/hal/lib/common/decimal_types_utils.h

```diff
@@ -0,0 +1,189 @@
+/*
```

Comment:
Can't we use default protobuf functions to create a Decimal64 message object?
Or create a utility function or macro to initialize the object?
`InitDecimal64(&val, digits, precision);`

Reply:
> Can't we use default protobuf functions to create a Decimal64 message object?

We can, and it leads to the boilerplate code:
```
::gmni::Decimal64* value = new Decimal64Value;
value->set_digits(1);
value->set_precision(2);
```

Anyway, if using this wrappers seems an overkill to you, we can use the naive approach. Please comment.

> Or create a utility function

This file contains such utility functions as `TypedDecimalInitializer` members. The reason they are put in the class is a compile-time type checking.

> or macro

Do we prefer using macros instead of template functions?

Reply:
I think native way to create the Decimal64 should be fine. No need to have another wrapper or matcher.